openedx/frontend-app-course-authoring
# feat: collections tab [FC-0062]

## Description

Adds collection cards to collections and home tabs.

## Supporting information

* https://github.com/openedx/frontend-app-course-authoring/issues/1102
* `Private-ref`: [FAL-3789](https://tasks.opencraft.com/browse/FAL-3789)

## Testing instructions

* Checkout https://github.com/openedx/edx-platform/pull/35321
* Setup tutor with this PR and https://github.com/open-craft/tutor-contrib-meilisearch plugin
* Make sure to create few libraries
* Run below snippet to add some collections to a library
```python
import random
from openedx_learning.api import authoring as authoring_api
from openedx_learning.api import authoring_models
learning_packages = authoring_models.LearningPackage.objects.all()
# set below learning_id to a specific library
learning_id = random.choice(learning_packages).id
for i in range(40):
    authoring_api.create_collection(
        learning_id,
        f"collection_{i}",
        None,
        description=f'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque et mi ac nisi accumsan imperdiet vitae at odio. Vivamus tempor nec lorem eget lacinia. Vivamus efficitur lacus non dapibus porta. Nulla venenatis luctus nisi id posuere. Sed sollicitudin magna a sem ultrices accumsan. Praesent volutpat tortor vitae luctus rutrum. Integer. Description {i}'
    )
```
* Run `tutor dev run cms ./manage.py cms reindex_studio --experimental`.
* Open the library in studio to which we added collections in above code snippet.
* Test infinite scrolling, search and sorting.

## Review notes:

* Please ignore commits: 227864cd58512f807d79a2ea4e47ae99937861cb and 192b67c5078ab697683bbed3c426e17816f58ed1 as they are part of https://github.com/openedx/frontend-app-course-authoring/pull/1263


## Concerns

* ~For now, I have excluded collections from `Recently modified` section completely. Let me know if that needs to be changed.~ Sam as [responded](https://github.com/openedx/frontend-app-course-authoring/issues/1102#issuecomment-2334393172) in the issue, have included collections in recently modified section.
* Component count under collection is missing in meilisearch index and backend.
* Tags are missing
* Collection is missing `key` field which is being used [here](https://github.com/open-craft/edx-platform/blob/6f397675e65a68c36817134340a4e938c2a0c59b/openedx/core/djangoapps/content/search/documents.py#L291). Had to revert back to id.
* Finally, the three button action menu in the collection card does nothing.
* `New:` Next page of components and collections are loaded together even if only one tab page is scrolled to bottom. I don't think that it is a bad thing as the data is fetched using a single api call to meilisearch. To test this, create multiple components as well as collections and scroll to bottom in one tab and load all items.
